### PR TITLE
[TT-4141] Implement new versioning migrator

### PR DESCRIPTION
**Pull request by furkansenharputlu**
This PR implements the migration function which migrates old versioning to new versioning fields in API definition by creating sub-version APIs and changing the base API accordingly.

### CODE REVIEW

File: apidef/migration.go [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+	"strings"
+)
+
+func (a *APIDefinition) MigrateVersioning() (versions []APIDefinition, err error) {
+	if a.VersionDefinition.Enabled || len(a.VersionDefinition.Versions) != 0 {
+		return nil, errors.New("not migratable - new versioning is enabled")
+	}
+
+	if a.VersionData.NotVersioned && len(a.VersionData.Versions) > 1 {
+		return nil, errors.New("not migratable - if not versioned, there should be 1 version info in versions map")
```
[the remark this answers is not in the document]
Comment (furkansenharputlu) [reply apidef/migration.go:15]: actually it must be just 1 version info, not at least.

[the remark this answers is not in the document]
Comment (furkansenharputlu) [reply apidef/migration.go:15]: If there are 2, we expect user to select one of them and remove other.

[the remark this answers is not in the document]
Comment (vaske) [reply apidef/migration.go:15]: ok cool

**Review by vaske — APPROVED** [review]

**Review by jeffy-mathew — APPROVED** [review]

**Pull request merged**